File: Selenium/Project/CRM_activity9.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeSuite;
import org.testng.annotations.Test;

import java.time.Duration;
import java.util.List;

public class CRM_activity9 {
        WebDriver driver = new FirefoxDriver();
        @BeforeSuite
        public void beforeClass() {
                System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
                WebDriverManager.firefoxdriver().setup();
                WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));
                driver.get("http://alchemy.hguy.co/crm");

        }

        @Test
        public void traverseTable() throws InterruptedException {
                WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));
                Actions builder = new Actions(driver);
                String pgelogin = driver.getTitle();
                System.out.println("The Page Title is: "+pgelogin);
                WebElement txtUname = driver.findElement(By.id("user_name"));
                WebElement txtPwd = driver.findElement(By.id("username_password"));
                WebElement btnLogin = driver.findElement(By.id("bigbutton"));
                txtUname.sendKeys("admin");
                txtPwd.sendKeys("pa$$w0rd");
                btnLogin.click();
                String pgeHome = driver.getTitle();
                System.out.println("The homePage title is: "+pgeHome);
                Assert.assertEquals(pgeHome, "SuiteCRM");
                WebElement sales = driver.findElement(By.id("grouptab_0"));
                wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//span[normalize-space()='My Open Cases']")));
//                sales.click();
                builder.moveToElement(sales).build().perform();
//                Thread.sleep(2000);
                WebElement leads = driver.findElement(By.id("moduleTab_9_Leads"));
                wait.until(ExpectedConditions.visibilityOf(leads));
                builder.moveToElement(leads).click().build().perform();
//                Thread.sleep(3000);
                System.out.println("*********** Navigating to Leads module *********");
//                Thread.sleep(3000);
                wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//h2[@class='module-title-text']")));

                for(int r=1; r<11;r++)
                {
                        String name = driver.findElement (By.xpath("//table[contains(@class,'responsive')]/tbody/tr["+r+"]/td[3]")).getText();
                        String user = driver.findElement (By.xpath("//table[contains(@class,'responsive')]/tbody/tr["+r+"]/td[8]")).getText();

                        System.out.println("Name displayed in Row number "+r+" is: "+name+" and respective User ID is: "+user);
                }

                System.out.println("Run Completed");
        }
        @AfterClass
        public void afterClass(){
                driver.quit();
        }
}

File: Selenium/Project/CRM_activity9.py
from selenium.webdriver.common.by import By
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC, expected_conditions

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # initialize the wait object
    wait = WebDriverWait(driver, 10)

    # Navigate to the URL
    driver.get("http://alchemy.hguy.co/crm")
    # Print the title of the page
    print("Page title is: ", driver.title)

    txtUname = driver.find_element(By.ID, "user_name")
    txtPwd = driver.find_element(By.ID, "username_password")
    btnLogin = driver.find_element(By.ID, "bigbutton")
    txtUname.send_keys("admin")
    txtPwd.send_keys("pa$$w0rd")
    btnLogin.click()

    pgeHome = driver.title
    print("The homePage title is: " + pgeHome)
    assert pgeHome == "SuiteCRM"

    sales = driver.findElement(By.ID, "grouptab_0")
    wait.until(expected_conditions.visibility_of_element_located(By.XPATH, "//span[normalize-space()='My Open Cases']"))

    # Set up the actions object
    actions = webdriver.ActionChains(driver)
    actions.move_to_element(sales).perform()

    leads = driver.findElement(By.ID, "moduleTab_9_Leads")
    wait.until(expected_conditions.visibility_of(leads))
    actions.move_to_element(leads).click().perform()

    print("*********** Navigating to Leads module *********")

    wait.until(expected_conditions.visibility_of_element_located(By.XPATH, "//h2[@class='module-title-text']"))

    for x in range(1,11):
        name = driver.findElement(By.XPATH, "//table[contains(@class,'responsive')]/tbody/tr[" + x + "]/td[3]").getText())
        user = driver.findElement(By.XPATH, "//table[contains(@class,'responsive')]/tbody/tr[" + x + "]/td[8]").getText())
        print("Name displayed in Row number " + x + " is: " + name + " and respective User ID is: " + user)

    print("Run Completed")

    # close the browser
    driver.quit()
